clustered/common/src/main/java/org/ehcache/clustered/common/messages/ServerStoreOpMessage.java

```diff
@@ -107,6 +165,17 @@ public ServerStoreOp operation() {
     public ByteBuffer getPayload() {
       return payload;
     }
+
+    @Override
+    public byte[] encode() {
+      byte[] encodedMsg = super.encode();
```

Comment:
I'm worried that the optimization would require to flatten the hierarchy. Let's not postpone that.